File: Práctica/2.5 - RobocodeLabo/robots/CasseraBottaBianchi/EstrategiaGanadora.java
package CasseraBottaBianchi;

public class EstrategiaGanadora implements RobotStrategy {
    private LaboRobot laboRobot;
    public EstrategiaGanadora(LaboRobot laboRobot) {
        this.laboRobot = laboRobot;
    }
    @Override
    public void run() {
		laboRobot.back(60); // Mueve el robot hacia atrás en píxeles
		laboRobot.turnGunRight(360); // Gira el arma a la derecha en grados.
		laboRobot.back(100); // Mueve el robot hacia atrás en píxeles
		laboRobot.turnGunLeft(360); // Gira el arma a la derecha en grados.
    }
    @Override
    public void onScannedRobot() {
        laboRobot.fire(2);
    }
    @Override
    public void onHitByBullet() {
        laboRobot.back(10);
        laboRobot.turnGunRight(360);
    }
    @Override
    public void onHitWall() {
        laboRobot.ahead(20);
    }
}

File: Talleres/3/src/EstrategiaGanadora.kt
package CasseraBottaBianchi

class EstrategiaGanadora(private val laboRobot: LaboRobot) : RobotStrategy {

    override fun run() {
        laboRobot.back(60) 
        laboRobot.turnGunRight(100) 
        laboRobot.back(100) 
        laboRobot.turnGunLeft(100) 
    }

    override fun onScannedRobot() {
        laboRobot.fire(2.0)
    }

    override fun onHitByBullet() {
        laboRobot.back(10)
    }

    override fun onHitWall() {
        laboRobot.ahead(20)
    }
}
